src/libraries/Native/Unix/System.Native/pal_process.c

```diff
@@ -203,6 +202,39 @@ static int SetGroups(uint32_t* userGroups, int32_t userGroupsLength, uint32_t* p
     return rv;
 }
 
+static int32_t SocketPair(int32_t sv[2])
+{
+    int32_t result;
+
+    int type = SOCK_STREAM;
+#ifdef SOCK_CLOEXEC
+    type |= SOCK_CLOEXEC;
+#endif
+
+    while ((result = socketpair(AF_UNIX, type, 0, sv)) < 0 && errno == EINTR);
+
+#ifndef SOCK_CLOEXEC
+    if (result == 0)
+    {
+        while ((result = fcntl(sv[0], F_SETFD, FD_CLOEXEC)) < 0 && errno == EINTR);
```

Comment:
For consistency sake, please use `READ_END_OF_PIPE` and `WRITE_END_OF_PIPE` instead of `0` and `1`